easybuild/tools/toolchain/toolchain.py

```diff
@@ -1140,21 +1139,21 @@ def _add_dependency_variables(self, names=None, cpp=None, ld=None):
         :names: list of strings containing the name of the dependency
         """
         # collect dependencies
-        dependencies = self.dependencies if names is None else [{"name": name} for name in names if name]
+        deps = self.dependencies if names is None else [{'name': name} for name in names if name]
 
-        # collect software install prefixes for dependencies
-        dependency_roots = []
-        for dep in dependencies:
-            if dep.get("external_module", False):
+        # collect software install prefixes for toolchain components + dependencies
+        dep_roots = []
+        for dep in deps + self.tcdeps:
```

Comment:
Just to point it out: here's the actual fix.

Everything else that was changed in `easybuild/*` is basically (very minor) code style improvement...